### PR TITLE
Add module_to_class/1 tests to beamtalk_stack_frame_tests (BT-2199)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

`beamtalk_stack_frame:module_to_class/1` is an exported function called by `beamtalk_test_case.erl` for test-output formatting and internally by `wrap_frame/1`. All three pattern-match branches (`"bt@"` prefix, `"beamtalk_"` prefix, fallback) plus the non-atom guard clause had **zero direct EUnit tests**.

This PR adds 9 focused test functions covering those branches without requiring any OTP application startup.

## Changes

- Added 9 EUnit tests to `runtime/apps/beamtalk_runtime/test/beamtalk_stack_frame_tests.erl`
  - `bt@` prefix: single-segment, multi-segment stdlib, alt format
  - `beamtalk_` prefix: single-word known atom, multi-word known atom, unknown atom (→ `nil`)
  - Fallback `_` case: `snake_to_class` returns `nil` (no registry call), known atom not registered (→ `nil`)
  - Non-atom guard clause

## Test Results

2113 tests, 0 failures (48 tests in the stack_frame module specifically).

## Linear

https://linear.app/beamtalk/issue/BT-2199/add-module-to-class1-tests-to-beamtalk-stack-frame-tests-0percent

https://claude.ai/code/session_01AJuhwMguTJQeemob4fhxjA

---
_Generated by [Claude Code](https://claude.ai/code/session_01AJuhwMguTJQeemob4fhxjA)_

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for module-name classification and conversion logic, validating multiple naming conventions, prefixed formats, unknown/edge cases, fallback behavior for unrecognized names, and type-safety guards to ensure predictable results for non-string inputs.

[Review Change Stack](https://app.coderabbit.ai/change-stack/jamesc/beamtalk/pull/2231?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: runtime/apps/beamtalk_runtime/test/beamtalk_stack_frame_tests.erl [hunk 1]
```diff
@@ -193,3 +193,59 @@ format_frame_file_no_line_test() ->
...
+%%% required because:
+%%%   - bt@ and beamtalk_ paths return snake_to_class directly, no registry.
+%%%   - The fallback nil-return path exits before the registry call.
+%%%   - erlang:whereis/1 returns undefined safely when no process is registered.
+%%% ===================================================================
+
+%% bt@ prefix, single-segment: 'bt@counter' → 'Counter'
+%% 'Counter' atom is in the table via tagged_map_tests.
+module_to_class_bt_prefix_single_word_test() ->
+    ?assertEqual('Counter', beamtalk_stack_frame:module_to_class('bt@counter')).
```
[the remark this answers is not in the document]
Comment (jamesc) [reply runtime/apps/beamtalk_runtime/test/beamtalk_stack_frame_tests.erl]: Fixed in e03717d — changed `'bt@counter'` → `'bt@stack_frame'` so the expected atom `'StackFrame'` is guaranteed by `make_test_frame/0` in this same file, eliminating the cross-file dependency.

---
_Generated by [Claude Code](https://claude.ai/code)_
